casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala

```diff
@@ -52,41 +56,127 @@ object CommUtil {
       _     <- TransportLayer[F].broadcast(peers, msg)
     } yield ()
 
+  def requestApprovedBlock[
+      F[_]: Monad: Capture: MultiParentCasperConstructor: Log: Time: Metrics: TransportLayer: NodeDiscovery: ErrorHandler: PacketHandler]
+    : F[Unit] = {
+    val request = ApprovedBlockRequest("PleaseSendMeAnApprovedBlock").toByteString
+
+    def askPeers(peers: List[PeerNode], local: PeerNode): F[Unit] = peers match {
+      case peer :: rest =>
+        for {
+          _    <- Log[F].info(s"CASPER: Sending request for ApprovedBlock to $peer")
+          send <- TransportLayer[F].roundTrip(peer, packet(local, request), 5.seconds)
+          _ <- send match {
+                case Left(err) =>
+                  Log[F].info(s"CASPER: Failed to get response from $peer because: $err") *>
+                    askPeers(rest, local)
+
+                case Right(response) =>
+                  Log[F]
+                    .info(s"CASPER: Received response from $peer! Processing...")
+                    .flatMap(_ => {
+                      val maybeSender = ProtocolHelper.sender(response)
+                      val maybePacket = toPacket(response).toOption
+
+                      (maybeSender, maybePacket) match {
+                        case (Some(sender), Some(_)) =>
+                          Connect
+                            .handlePacket[F](sender, maybePacket)
+                            .flatMap(_ => {
+                              MultiParentCasperConstructor[F].lastApprovedBlock.flatMap {
+                                case Some(_) => ().pure[F] //valid ApprovedBlock received
+                                case None    => askPeers(rest, local)
+                              }
+                            })
+                        case _ => Log[F].info(s"CASPER: Response invalid.") *> askPeers(rest, local)
```

Comment:
this should not be an `info`, it shouyld be logged as `error`. also please include what went wrong, if sender was available, what was the mesage type